gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufdataset.cpp

```diff
@@ -179,18 +180,21 @@ OGRFlatGeobufDataset::~OGRFlatGeobufDataset()
 
 GDALDataset *OGRFlatGeobufDataset::Open(GDALOpenInfo* poOpenInfo)
 {
-    if( OGRFlatGeobufDriverIdentify(poOpenInfo) == FALSE ||
-        poOpenInfo->eAccess == GA_Update )
-    {
+    if (OGRFlatGeobufDriverIdentify(poOpenInfo) == FALSE)
         return nullptr;
-    }
 
     const auto bVerifyBuffers = CPLFetchBool( poOpenInfo->papszOpenOptions, "VERIFY_BUFFERS", true );
 
+    auto isDir = CPL_TO_BOOL(poOpenInfo->bIsDirectory);
+    auto bUpdate = poOpenInfo->eAccess == GA_Update;
+
+    if (isDir && bUpdate){
+        CPLError(CE_Failure, CPLE_AppDefined, "Unable to open dir in update mode");
```

Comment:
I suspect this CPLError() is causing the CI issues in the FileGDB one. When a driver emits a CPLError() in its Open() code, no other drivers are tried afterwards. And as OGRFlatGeobufDriverIdentify() just returns -1 "I may recognize or not this dataset" if a directory is passed, this will go here. So probably just remove the CPLError()